launcher/tools/GenericProfiler.cpp

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2023 Trial97 <alexandru.tripon97@gmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "GenericProfiler.h"
+
+#include "BaseInstance.h"
+#include "launch/LaunchTask.h"
+#include "settings/SettingsObject.h"
+
+class GenericProfiler : public BaseProfiler {
+    Q_OBJECT
+   public:
+    GenericProfiler(SettingsObjectPtr settings, InstancePtr instance, QObject* parent = 0);
+
+   protected:
+    void beginProfilingImpl(shared_qobject_ptr<LaunchTask> process);
+};
+
+GenericProfiler::GenericProfiler(SettingsObjectPtr settings, InstancePtr instance, QObject* parent)
+    : BaseProfiler(settings, instance, parent)
+{}
+
+void GenericProfiler::beginProfilingImpl(shared_qobject_ptr<LaunchTask> process)
+{
+    emit readyToLaunch(tr("Started process: %1").arg(process->pid()));
```

Comment:
isn't this already shown?

Reply:
If you mean the text no the method is a pure virtual one. So I need to overload it

Reply:
I haven't looked at in detail. but i don't think this profiler needs to print anything so maybe you should find a way to.. not.

Reply:
It should in order to let the user open it's own profiler/debugger to attach to the process

Reply:
![image](https://github.com/PrismLauncher/PrismLauncher/assets/57493648/ab9b8686-f26d-41cc-b2b4-2a487aad8b17)

Reply:
Yea, I know that it's in logs what I mean is that I still need a dialog box to display to the user. 
Why not use that to also display the process ID?
If I remove that the message will be something like this:
```suggestion
    emit readyToLaunch(tr("Process started"));
```

Reply:
Oh, nvm lol